src/components/RadonQuery.vue

```diff
@@ -1,286 +1,31 @@
-            </div>
-          </div>
-          <div>
-            <RadonScript
-              class="script"
-              v-show="!error.retrieve"
-              :path="{stage: 'retrieve', retrieveIndex: index}"
-              :script="source.script"
-            />
-            <p class="error" v-show="error.retrieve">There is an error in the retrieve stage</p>
-          </div>
-        </div>
-        <div>
-          <Button class="add-source" type="dashed" :onClick="pushRetrieve">+ ADD SOURCE</button>
-        </div>
-      </div>
-      <p class="name">Retrieval</p>
-    </div>
-
-    <div class="stage">
-      <RadonScript
-        class="script"
-        v-show="!error.aggregate"
-        :path="{stage: 'aggregate'}"
-        :script="aggregate.script"
-      />
-      <p class="error" v-show="error.aggregate">There is an error in the aggregate stage</p>
-      <p class="name">Aggregation</p>
-    </div>
-
-    <div class="stage">
-      <RadonScript
-        class="script"
-        v-show="!error.consensus"
-        :path="{stage: 'consensus'}"
-        :script="consensus.script"
-      />
-      <p class="error" v-show="error.consensus">There is an error in the consensus stage</p>
-      <p class="name">Consensus</p>
-    </div>
-    <div class="stage">
-      <input class="deliver" placeholder="url">
-        <p class="name">Deliver</p>
-      <div class>
-      </div>
-    </div>
-    <div>
-      <p>Retrieval</p>
-      <textarea
-        class="textarea"
-        :value="JSON.stringify(retrieve)"
-        v-on:keyup.enter="e => updateStage(e, 'retrieve')"
-      />
-      <p>Aggregation</p>
-      <textarea
-        class="textarea"
-        :value="JSON.stringify(aggregate)"
-        v-on:keyup.enter="e => updateStage(e, 'aggregate')"
-      />
-      <p>Consensus</p>
-      <textarea
-        class="textarea"
-        :value="JSON.stringify(consensus)"
-        v-on:keyup.enter="e => updateStage(e, 'consensus')"
-      />
-
-      <button @click="tryDataRequest">Try data request</button>
-    </div>
-  The result of the data request is:
-  {{ radRequestResult }}
-
+  <div>
+    <RadonStage class="stage" :retrieve="retrieve" />
+    <RadonStage class="stage" :aggregate="aggregate" />
+    <RadonStage class="stage" :consensus="consensus" />
   </div>
 </template>
 
 <script>
 import { mapState } from 'vuex'
 
-import RadonScript from '@/components/RadonScript.vue'
-import Button from '@/components/Button.vue'
-
-import {
-  FilteringFunctionCodes,
-  HashFunctionCodes,
-  ReducingFunctionCodes,
-} from '@/radon'
-
+import RadonStage from '@/components/RadonStage.vue'
 export default {
   name: 'RadonQuery',
   components: {
-    RadonScript,
-    Button,
-  },
-  props: {
-    radRequest: Object,
-  },
-  data () {
-    return {
-      error: {
-        retrieve: false,
-        aggregate: false,
-        consensus: false,
-      },
-    }
+    RadonStage,
   },
```

Comment:
It would be nice to have this method in the toolbar and delete this component so we can include these features directly in the editor.